File: springwebfluxtutorial/src/main/java/com/qiusuo/webfluxtutorial/repository/impl/CustomUserRepositoryImpl.java
package com.qiusuo.webfluxtutorial.repository.impl

import com.qiusuo.webfluxtutorial.domain.User;
import com.qiusuo.webfluxtutorial.repository.CustomUserRepository;
import org.springframework.data.r2dbc.core.DatabaseClient;
import reactor.core.publisher.Flux;

public class CustomUserRepositoryImpl implements CustomUserRepository {
    private DatabaseClient databaseClient;

    public CustomUserRepositoryImpl(DatabaseClient databaseClient) {
    }

    public Flux<User> findAllUsers() {
        return Flux.empty().ofType(User.class);
    }
}

File: springwebfluxtutorial/src/main/kotlin/com/qiusuo/webfluxtutorial/repository/impl/CustomUserRepositoryImpl.kt
package com.qiusuo.webfluxtutorial.repository.impl

import com.qiusuo.webfluxtutorial.domain.User
import com.qiusuo.webfluxtutorial.repository.CustomUserRepository
import org.springframework.r2dbc.core.DatabaseClient
import reactor.core.publisher.Flux

class CustomUserRepositoryImpl(private val databaseClient: DatabaseClient) : CustomUserRepository {
    override fun findAllUsers(): Flux<User> {
        return Flux.empty<Any>().ofType(User::class.java)
    }
}